tests/framework/installer/ceph_installer.go

```diff
@@ -96,47 +89,46 @@ func (h *CephInstaller) CreateCephOperator(namespace string) (err error) {
 		assert.NoError(h.T(), err)
 	}
 
-	err = h.k8shelper.CreateNamespace(namespace)
-	if err != nil {
-		if kerrors.IsAlreadyExists(err) {
-			logger.Warningf("Namespace %q already exists!!!", namespace)
-		} else {
-			return fmt.Errorf("failed to create namespace %q. %v", namespace, err)
-		}
+	// The operator namespace needs to be created explicitly, while the cluster namespace is created with the common.yaml
+	if err := h.k8shelper.CreateNamespace(h.settings.OperatorNamespace); err != nil {
+		return err
 	}
 
-	// disable admission controller for Kubernetes version older than v1.16.0 and for upgrade test as api version v1 require minimum v0.7 controller runtime and upgrade test still using
-	// older version of controller runtime.
-	if !utils.IsPlatformOpenShift() && namespace != "upgrade-ns-system" {
-		err = h.startAdmissionController(namespace)
-		if err != nil {
-			return fmt.Errorf("Failed to start admission controllers: %v", err)
-		}
+	// Create the namespace and RBAC before starting the operator
+	_, err = h.k8shelper.KubectlWithStdin(h.Manifests.GetCommon(), createFromStdinArgs...)
+	if err != nil {
+		return fmt.Errorf("Failed to create rook-operator pod: %v ", err)
 	}
 
-	rookOperator := h.Manifests.GetRookOperator(namespace)
-	_, err = h.k8shelper.KubectlWithStdin(rookOperator, createFromStdinArgs...)
+	err = h.startAdmissionController()
 	if err != nil {
-		return fmt.Errorf("Failed to create rook-operator pod: %v ", err)
+		return fmt.Errorf("Failed to start admission controllers: %v", err)
 	}
 
-	logger.Infof("Rook Operator started")
+	_, err = h.k8shelper.KubectlWithStdin(h.Manifests.GetOperator(), createFromStdinArgs...)
+	if err != nil {
+		return fmt.Errorf("Failed to create rook-operator pod: %v", err)
+	}
 
+	logger.Infof("Rook operator started")
 	return nil
 }
 
-func (h *CephInstaller) startAdmissionController(namespace string) error {
-	err := h.k8shelper.CreateNamespace(namespace)
-	if err != nil {
-		if kerrors.IsAlreadyExists(err) {
-			logger.Warningf("Namespace %q already exists!!!", namespace)
-		} else {
-			return fmt.Errorf("failed to create namespace %q. %v", namespace, err)
-		}
-	}
+func (h *CephInstaller) startAdmissionController() error {
 	if !h.k8shelper.VersionAtLeast("v1.16.0") {
+		logger.Info("skipping the admission controller on K8s version older than v1.16")
 		return nil
 	}
+	if !h.settings.EnableAdmissionController {
+		logger.Info("skipping admission controller for this test suite")
+		return nil
+	}
+	// disable admission controller for Kubernetes version older than v1.16.0
+	if utils.IsPlatformOpenShift() {
+		logger.Info("skipping the admission controller on OpenShift")
```

Comment:
Out of curiosity, why?

Reply:
As I recall there was some issue with the admission controller script that still needs to be fixed for openshift. After we actually get the tests running regularly there, we should fix this.

Reply:
Yes, when I added this at that time admission controller was not starting(I haven't check recently) on openshift so I skipped after discussing.